12-watchtower-API.md

```diff
@@ -127,122 +129,167 @@ This message contains all the information regarding the appointment that the cli
 
-### Rationale
+### The `appointment_accepted` message
 
-We can group the data fields into logical groups. 
+This message contains information about the acceptance of an appointment from the WatchTower.
 
-* **Appointment information**: The appointment time is defined using the ```start_block``` and```end_block```. WatchTower will delete the job when the appointment has expired. We recommend only using block numbers as that is the natural clock for Bitcoin. 
-* **Explicit acknowledgement of transaction details**: The ```dispute_delta```, ```transaction_size```, and ```transaction_fee``` let the WatchTower confirm the transaction is "reasonable" and it can be accepted into the blockchain (especially if there is congestion in the future). 
-* **Encrypted transaction**: The ```cipher```,```encrypted_blob``` states how the WatchTower can later find the dispute transaction and decrypt the justice transaction. 
-* **Customer signature** The ```customer_public_key``` and ```customer_signature``` provides an explicit message about the job from the customer. 
+1. type: ? (`appointment_accepted `)
+2. data:
+   * [`16*byte `:`locator`]
+   * [`varsize`: `op_receipt`]
+   * [`u16`: `op_wt_signature_algorithm`]
+   * [`varsize`: `op_wt_signature`]
+   * [`varsize`: `op_wt_public_key`]
 
-Generally, this standard is trying to allow a reputationally accountable watching service. The signed job from the customer provides an explicit acknowledgement of the transaction details that is important for the WatchTower to decide whether they can accept it. If the decrypted justice transaction does not satisfy the signed job (e.g. fee too low), then the WatchTower is not obliged to fulfil it. 
+The sending node:
 
-The _explictiness_ of the signed job ensures there is a clear protocol trasncript between the customer and WatchTower. Given the blockchain and decrypted justice transaction, anyone can verify that the WatchTower could have satisified the job. 
+* MUST receive `appointment` before sending an `appointment_accepted` message.
+* MUST set the `locator` to match the one received in `appointment`.
+* if `qos` was agreed on `wt_init`:
+	* MUST set `op_receipt`] to a receipt build according to 	 [Signed-Receipt](#signed-receipt)
+	* MUST set `op_wt_signature_algorithm` to one of the signature algorithms agreed on `wt_init`.
+	* MUST set `op_wt_signature` to the signature of the appointment using `op_customer_signature_algorithm`.
+	* MUST set `op_wt_public_key` to the public key that matches the private key used to create `op_customer_signature`.
+
+The receiving node:
+
+* MUST fail the connection if:
+	* `locator` does not match any of the previously sent to the WatchTower.
+
+* if `qos` was agreed on `wt_init`:
+The receiving node MUST also reject the appointment if:
```

Comment:
i wonder what rejection means here. its too late since the watchtower will deduct its balance. does it just notify user that watchtowers receipt was invalid? or was not satisfactory?

Reply:
It depends on the type of payment. If you're doing subscription based for example, it doesn't have to. You just delete the data and do not count that as an appointment.

It's a similar to the situation (not described in the BOLT) were you're expecting a payment and the user sends you the data without paying. You'll reject the appointment.

Reply:
ya, but the point here is that the client / customer is rejecting the signed appointment from the watchtower. if it rejects it, what does it do? is it worth making notes about this situation?

Reply:
It says fail the connection. If you get a random 'appointment_accepted' from a connection someone has stablished to you, you just drop the connection.

Reply:
Or if the watchtower sends you an `appointment_accepted` that does not belong to you, you disconnect from the tower.

Reply:
ok but if it gets rejected what should happen. is it worth making notes that the user should try to hire another watchtower? re-attempt the job? should we give suggestions here?

Reply:
No, the protocol should not have loops. It's up to the implementation how to handle UX (check other BOLTs for that).

Reply:
ya should do we make suggestions about how to resolve? or just leave it blank